fix: add CLI argument `browser-path`

jstrachan opened the pull request:
**What this PR does / why we need it**:

adds a CLI argument:

* `browser-path` so you can specify the initial context path to open the browser as you start Octant.

**Release note**:
```

This release adds a new CLI argument:

* `browser-path` so you can specify the initial context path to open the browser as you start Octant

e.g. to open octant on the Deployments view:

    octant --browser-path "/#/overview/namespace/default/workloads/deployments"

```

wwitzel3 commented on Makefile:
```diff
@@ -0,0 +1,16 @@
+
```
remove this file

jstrachan replied:
doh - can't believe I added it back again ;) sorry about that...

jstrachan replied:
@wwitzel3 I just pushed an updated PR without that pesky Makefile ;)

wwitzel3 reviewed (changes requested):
Looks good, just need to remove that Makefile from the PR.

jstrachan committed:
fix: add CLI argument `browser-path`

so you can specify the initial context path to open the browser as you start Octant

Signed-off-by: James Strachan <james.strachan@gmail.com>

wwitzel3 reviewed (approved):
LGTM, I want one more set of eyes on it from @vmware-tanzu/octant-reviewers to makes sure there isn't something I'm not thinking about.

GuessWhoSamFoo commented:
Not a blocker for this PR, but what should happen if there is a conflict between the URL path and default namespace?

jstrachan commented:
> Not a blocker for this PR, but what should happen if there is a conflict between the URL path and default namespace?

Not sure I follow; you mean what if the URL path is for a namespace that doesn't exist? If so there's nothing stopping folks typing the same invalid URL into a browser address bar too right? Hopefully Octant can report if a namespace doesn't exist.

GuessWhoSamFoo commented:
Octant will notify users of an invalid namespace.

There's a namespace flag that assumes starting on the root path. In this case, the URL path will win. I wanted to point out there can be flags with conflicting user intent.
